lib/srv/server/ssm_install_test.go

```diff
@@ -152,17 +162,117 @@ func TestSSMInstaller(t *testing.T) {
+				},
+				DocumentName: document,
+				Params:       map[string]string{"token": "abcdefg"},
+				SSM: &mockSSMClient{
+					commandOutput: &ssm.SendCommandOutput{
+						Command: &ssm.Command{
+							CommandId: aws.String("command-id-1"),
+						},
+					},
+					invokeOutput: &ssm.GetCommandInvocationOutput{
+						Status:       aws.String(ssm.CommandStatusSuccess),
+						ResponseCode: aws.Int64(0),
+					},
+					describeOutput: &ssm.DescribeInstanceInformationOutput{
+						InstanceInformationList: []*ssm.InstanceInformation{
+							{
+								InstanceId:   aws.String("instance-id-1"),
+								PingStatus:   aws.String("Online"),
+								PlatformType: aws.String("Linux"),
+							},
+							{
+								InstanceId:   aws.String("instance-id-2"),
+								PingStatus:   aws.String("ConnectionLost"),
+								PlatformType: aws.String("Linux"),
+							},
+							{
+								InstanceId:   aws.String("instance-id-3"),
+								PingStatus:   aws.String("Online"),
+								PlatformType: aws.String("Windows"),
+							},
+						},
+					},
+				},
+				Region:    "eu-central-1",
+				AccountID: "account-id",
+			},
+			conf: SSMInstallerConfig{
+				Emitter: &mockEmitter{},
+			},
+			expectedEvents: []events.AuditEvent{
+				&events.SSMRun{
+					Metadata: events.Metadata{
+						Type: libevent.SSMRunEvent,
+						Code: libevent.SSMRunSuccessCode,
+					},
+					CommandID:  "command-id-1",
+					InstanceID: "instance-id-1",
+					AccountID:  "account-id",
+					Region:     "eu-central-1",
+					ExitCode:   0,
+					Status:     ssm.CommandStatusSuccess,
+				},
+				&events.SSMRun{
+					Metadata: events.Metadata{
+						Type: libevent.SSMRunEvent,
+						Code: libevent.SSMRunFailCode,
+					},
+					CommandID:  "no-command",
+					InstanceID: "instance-id-2",
+					AccountID:  "account-id",
+					Region:     "eu-central-1",
+					ExitCode:   -1,
+					Status:     "SSM Agent in EC2 Instance is not connecting to SSM Service. Restart or reinstall the SSM service. See https://docs.aws.amazon.com/systems-manager/latest/userguide/ami-preinstalled-agent.html#verify-ssm-agent-status for more details.",
+				},
+				&events.SSMRun{
+					Metadata: events.Metadata{
+						Type: libevent.SSMRunEvent,
+						Code: libevent.SSMRunFailCode,
+					},
+					CommandID:  "no-command",
+					InstanceID: "instance-id-3",
+					AccountID:  "account-id",
+					Region:     "eu-central-1",
+					ExitCode:   -1,
+					Status:     "EC2 instance is running an unsupported Operating System. Only Linux is supported.",
+				},
+				&events.SSMRun{
+					Metadata: events.Metadata{
+						Type: libevent.SSMRunEvent,
+						Code: libevent.SSMRunFailCode,
+					},
+					CommandID:  "no-command",
+					InstanceID: "instance-id-4",
+					AccountID:  "account-id",
+					Region:     "eu-central-1",
+					ExitCode:   -1,
+					Status:     "EC2 Instance is not registered in SSM. Make sure that the instance has AmazonSSMManagedInstanceCore policy assigned.",
```

Comment:
Suggestion: matching only part of the status, as that's typically more resilient to future changes.

Reply:
I think it's fine given this is an audit log message which users can expect it to be stable.